src/control/cmd/dmg/pool.go

```diff
@@ -204,6 +235,122 @@ func (cmd *PoolCreateCmd) Execute(args []string) error {
 	return nil
 }
 
+// XXX DAOS-9196 Some extra bytes are needed for the metadata of the pool: at least 135 MB
+// are needed.  The extraByes is a little bigger than this limit, because the size of the
+// metadata will eventually grow along the life of the pool.
+const PoolMetadataBytes uint64 = uint64(200)*uint64(humanize.MiByte)
+
+// Return the maximal size of a pool which could be created with all the storage nodes
+func (cmd *PoolCreateCmd) GetMaxPoolSize(ctx context.Context) (scmBytes uint64,
+                                                               nvmeBytes uint64,
+                                                               errOut error) {
+	storageScanReq := &control.StorageScanReq{Usage: true}
+	resp, err := control.StorageScan(ctx, cmd.ctlInvoker, storageScanReq)
+	if err != nil {
+		errOut = err
+		return
+	}
+
+	if len(resp.HostStorage) <= 0 {
+		return 0, 0, errors.New("No DAOS server available")
+	}
+
+	hostStorageMap := resp.HostStorage
+	scmBytes = math.MaxUint64
+	nvmeBytes = math.MaxUint64
+	for _, key := range hostStorageMap.Keys() {
+		hostStorageSet := hostStorageMap[key]
+
+		hostSet := hostStorageSet.HostSet
+		if hostSet.Count() != 1 {
+			// XXX DAOS-9196 After code investigation, the HostSet should always
```

Comment:
From what I understand this multiple hostset should never happen. However, I would like to keep this test as HostStorage response could be agregated in the Future. Moreover, Multiple host should also be properly managed as we are just looking for minimal storage values. Thus, what I propose is to remove the panic with a warning message.
Could tell me, if this solution is acceptable for you ?

Reply:
Yes, you are unlikely to get aggregated results (hostset representing more than one host) with `Usage: true` set in the request with the current server side implementation. If your implementation relies upon responses not being aggregated then you should return an error in the case that aggregated responses are received so that when other areas of implementation change it triggers errors and the functional regressions are captured.

Reply:
At the end, as my implementation is independent if the host are aggregated or not,  I have totally remove this useless code. I also updated the test to check that agregated hostset are properly handle.